packages/components/dist/alpheios-components.js

```diff
@@ -66906,41 +66915,47 @@ class LexicalQuery extends _query_js__WEBPACK_IMPORTED_MODULE_1__["default"] {
 
     yield 'Retrieval of lemma translations completed'
 
-    const adapterLexiconResShort = yield alpheios_client_adapters__WEBPACK_IMPORTED_MODULE_3__["ClientAdapters"].lexicon.alpheios({
-      method: 'fetchShortDefs',
-      clientId: this.clientId,
-      params: {
-        opts: lexiconShortOpts,
-        homonym: this.homonym,
-        callBackEvtSuccess: LexicalQuery.evt.SHORT_DEFS_READY,
-        callBackEvtFailed: LexicalQuery.evt.SHORT_DEFS_NOT_FOUND
+    if (!this.startedDefinitionsQueries) {
```

Comment:
the problem with this is that we sometimes DO want to run the definitions queries a 2nd time. We do this crazy 2nd iteration with the persian lookup because the persian morphological parser is a little limited and the dictionary does support some full forms as well as lemmas. So, we first lookup to see if there is an entry for the full form of the persian word, and if we don't get results, then we try again using the morphological parser to parse the lemma and then we try looking up the lemma. Often it will be the same as the full form, but occasionally the parser adds some value here and the 2nd lookup in the dictionary is for a different word than the first and we get a result.

Eventually I think we need to move all of this business logic behind looking up words to a separate service, probably behind a GraphQL api so that we can control and fine tune it more easily. 

In the meantime, maybe checking to see if we started queries *for the same homonym already* is what we need to do.

Reply:
I didn't think that we have such a specific workflow for Persian inside LexicalQuery workflow.
Ok - will add check for targetWord.
The main problem here - that getting data from Persian is too long - and this leads to multiple upload from Persian. May be it is worth to add check for multiple request to client-adapters.